x-pack/plugins/gis/public/shared/layers/layer.js

```diff
@@ -139,21 +142,20 @@ export class ALayer {
     return this._source.renderSourceSettingsEditor({ onChange });
   };
 
+  getSourceDataRequest() {
+    return this._dataRequests.find(dataRequest => dataRequest.getDataId() === 'source');
+  }
+
   isLayerLoading() {
     return this._dataRequests.some(dataRequest => dataRequest.isLoading());
   }
 
-  dataHasLoadError() {
-    return this._dataRequests.some(dataRequest => dataRequest.hasLoadError());
+  hasErrors() {
+    return _.get(this._descriptor, 'isInErrorState', false);
```

Comment:
Is `isInErrorState` needed? Why not just check if this._descriptor.errorMessage exists or not?

Reply:
Not needed, just preferred